### PR TITLE
JAVA-18602: update spring-retry version

### PR DESCRIPTION
**Pull request by ehsansasanian**

### CODE REVIEW

File: spring-scheduling/src/main/java/com/baeldung/async/AsyncComponent.java [hunk 1]
```diff
@@ -20,9 +20,7 @@ public Future<String> asyncMethodWithReturnType() {
         try {
             Thread.sleep(5000);
             return new AsyncResult<>("hello world !!!!");
-        } catch (final InterruptedException e) {
-
-        }
+        } catch (final InterruptedException ignored) {}
```
Comment (dkapil) [inline spring-scheduling/src/main/java/com/baeldung/async/AsyncComponent.java:23]: revert this change as it is not different from the existing one

**Review by dkapil — CHANGES_REQUESTED** [review]

File: spring-scheduling/src/main/java/com/baeldung/springretry/MyServiceImpl.java [hunk 2]
```diff
@@ -20,23 +20,23 @@ public void retryService() {
 
     @Override
     public void retryServiceWithRecovery(String sql) throws SQLException {
-        if (StringUtils.isEmpty(sql)) {
+        if (!StringUtils.hasLength(sql)) {
```
[the remark this answers is not in the document]
Comment (ehsansasanian) [reply spring-scheduling/src/main/java/com/baeldung/springretry/MyServiceImpl.java:23]: StringUtils.isEmpty is deprecated. hasLength is an alternative for that. Should I still revert this one as well?
image

File: spring-scheduling/src/main/java/com/baeldung/async/AsyncComponent.java (hunk 1 above)
Comment (ehsansasanian) [reply spring-scheduling/src/main/java/com/baeldung/async/AsyncComponent.java:23]: will do this.

File: spring-scheduling/src/main/java/com/baeldung/springretry/AppConfig.java [hunk 3]
```diff
@@ -20,7 +20,7 @@ public RetryTemplate retryTemplate() {
         RetryTemplate retryTemplate = new RetryTemplate();
 
         FixedBackOffPolicy fixedBackOffPolicy = new FixedBackOffPolicy();
-        fixedBackOffPolicy.setBackOffPeriod(2000l);
+        fixedBackOffPolicy.setBackOffPeriod(2000L);
```
[the remark this answers is not in the document]
Comment (ehsansasanian) [reply spring-scheduling/src/main/java/com/baeldung/springretry/AppConfig.java:23]: it's not necessary, will revert this one as well.

File: spring-scheduling/src/main/java/com/baeldung/springretry/MyServiceImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dkapil) [reply spring-scheduling/src/main/java/com/baeldung/springretry/MyServiceImpl.java:23]: Can, we use Apache Commons String Utils https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isEmpty-java.lang.CharSequence- if that library is available in this module, else let's go with the hasLength change

[the remark this answers is not in the document]
Comment (ehsansasanian) [reply spring-scheduling/src/main/java/com/baeldung/springretry/MyServiceImpl.java]: Done
